app/src/shared/web3/web3Poller/web3Poller.js

```diff
@@ -0,0 +1,185 @@
+                    this.emitter.emit(events.ACCOUNT_ERROR, err)
+                }
+            })
+    )
+
+    handleAccount = (account: string) => {
+        const next = account
+
+        const didChange = !!(this.account && next && !areAddressesEqual(this.account, next))
+        const didDefine = !!(!this.account && next)
+
+        // Check current provider so that account event is not sent prematurely
+        // (ie. wait for user to approve access to Metamask)
+        if (this.web3.currentProvider !== null && (didDefine || didChange)) {
+            this.account = next
+            this.emitter.emit(events.ACCOUNT, next)
+        }
+    }
+
+    fetchChosenEthereumNetwork = () => (
+        this.web3.getEthereumNetwork()
+            .then((network) => {
+                this.handleNetwork(network.toString())
+            }, () => {
+                this.web3.getDefaultAccount()
+                    .catch((err) => {
+                        if (this.account !== null) {
+                            this.emitter.emit(events.NETWORK_ERROR, err)
+                        }
+                    })
+            })
+    )
+
+    handleNetwork = (network: NumberString) => {
+        const next = network
+        const didChange = this.networkId && next && this.networkId !== next
+        const didDefine = !this.networkId && next
+        if (didDefine || didChange) {
+            this.networkId = next
+            this.emitter.emit(events.NETWORK, next)
+        }
+    }
+
+    startPendingTransactionsPoll = () => {
+        this.clearPendingTransactionsPoll()
+        this.pendingTransactionsPollTimeout = setTimeout(this.pollPendingTransactions, PENDING_TX_POLL_INTERVAL)
+    }
+
+    pollPendingTransactions = () => (
+        this.handlePendingTransactions()
+            .then(this.startPendingTransactionsPoll, this.startPendingTransactionsPoll)
+    )
+
+    handlePendingTransactions = () => {
+        const web3 = getPublicWeb3()
+
+        return Promise.all(Object.keys(getTransactionsFromSessionStorage()).map(async (txHash) => {
+            const completed = await hasTransactionCompleted(txHash)
+            if (completed) {
+                const receipt = await web3.eth.getTransactionReceipt(txHash)
+
+                // Cannot trust that receipt won't be null... the next interval should receive it
+                if (receipt) {
+                    if (receipt.status === true) {
+                        this.emitter.emit(
+                            events.TRANSACTION_COMPLETE,
+                            txHash,
+                            receipt,
+                        )
+                    } else {
+                        this.emitter.emit(
+                            events.TRANSACTION_ERROR,
+                            txHash,
+                            new TransactionError(I18n.t('error.txFailed'), receipt),
+                        )
+                    }
+                }
+            }
+        }))
```

Comment:
you may also want to emit transaction `TRANSACTION_ERROR` if any of the async calls in this function happen to error e.g. `getTransactionReceipt`, or it might be fine for such an error to bubble up/crash? i.e. `try/catch` around the entire function with a `this.emitter.emit(events.TRANSACTION_ERROR, …)` in the catch, though this will emit a new error if any of the event handlers throw… so consider the following:
```js
    return Promise.all(Object.keys(getTransactionsFromSessionStorage()).map(async (txHash) => {
        let completed
        let receipt
        try {
            completed = await hasTransactionCompleted(txHash)
            receipt = !!completed && await web3.eth.getTransactionReceipt(txHash)
        } catch (err) {
            console.warn(err) // log unexpected errors
            this.emitter.emit(
                events.TRANSACTION_ERROR,
                txHash,
                new TransactionError(I18n.t('error.txFailed')), // could perhaps just pass the caught err or maybe TransactionError should know how to wrap another error?
            )
            return // bail out
        }
        if (completed && receipt) {
            // Cannot trust that receipt won't be null... the next interval should receive it
            if (receipt.status === true) {
                this.emitter.emit(
                    events.TRANSACTION_COMPLETE,
                    txHash,
                    receipt,
                )
            } else {
                this.emitter.emit(
                    events.TRANSACTION_ERROR,
                    txHash,
                    new TransactionError(I18n.t('error.txFailed'), receipt),
                )
            }
        }
    }))
```

Reply:
The error in this case is a bit misleading, the emitted error should reflect the transaction failing or succeeding in the blockchain. But I've added the logging at least if something happens here.